ext/random/randomizer.c

```diff
@@ -150,6 +150,80 @@ PHP_METHOD(Random_Randomizer, nextFloat)
 }
 /* }}} */
 
+static double getFloat_gamma_low(double x)
+{
+	return x - nextafter(x, -DBL_MAX);
+}
+
+static double getFloat_gamma_high(double x)
+{
+	return nextafter(x, DBL_MAX) - x;
+}
+
+static double getFloat_gamma(double x, double y)
```

Comment:
I believe that the cost of a call to `nextafter()` is often sufficiently high that testing for `x` and `y` before calling `getFloat_gamma_low()` and `getFloat_gamma_high()` entails (slightly) better performances:
```C
static double getFloat_gamma(double x, double y)
{
	return (fabs(x) > fabs(y)) ? getFloat_gamma_high(x) : getFloat_gamma_low(y);
}
```